game/src/sandbox/speed.rs

```diff
@@ -156,34 +159,132 @@ impl TimePanel {
-                    .centered_vert()
-                    .align_right()
+                    None
+                }
+            });
+        let baseline_finished_width: Option<f64> = baseline_finished_ratio
+            .map(|baseline_finished_ratio| baseline_finished_ratio * bar_width);
+
+        let cursor_width = 2.0;
+        let mut progress_bar = GeomBatch::new();
+
+        {
+            // TODO Why is the rounding so hard? The white background is always rounded
+            // at both ends. The moving bar should always be rounded on the left, flat
+            // on the right, except at the very end (for the last 'radius' pixels). And
+            // when the width is too small for the radius, this messes up.
+            progress_bar.push(bar_bg, Polygon::rectangle(bar_width, bar_height));
+            progress_bar.push(bar_fg, Polygon::rectangle(finished_width, bar_height));
+
+            if let Some(baseline_finished_width) = baseline_finished_width {
+                if baseline_finished_width > 0.0 {
+                    let baseline_cursor = Polygon::rectangle(cursor_width, bar_height)
+                        .translate(baseline_finished_width, 0.0);
+                    progress_bar.push(cursor_fg, baseline_cursor);
+                }
+            }
+        }
+
+        let text_geom = Text::from(
+            Line(format!("Finished Trips: {}", prettyprint_usize(finished),)).fg(text_color),
+        )
+        .render(ctx)
+        .translate(8.0, 0.0);
+        progress_bar.append(text_geom);
+
+        if let Some(baseline_finished_width) = baseline_finished_width {
+            let triangle_width = 9.0;
+            let triangle_height = 9.0;
+
+            // Add a triangle-shaped cursor above the baseline cursor
+            progress_bar = progress_bar.translate(0.0, triangle_height);
+
+            use geom::Triangle;
+            let triangle = Triangle::new(
+                Pt2D::zero(),
+                Pt2D::new(triangle_width, 0.0),
+                Pt2D::new(triangle_width / 2.0, triangle_height),
+            );
+            let mut triangle_poly = Polygon::from_triangle(&triangle);
+            triangle_poly = triangle_poly.translate(
+                baseline_finished_width - triangle_width / 2.0 + cursor_width / 2.0,
+                0.0,
+            );
+
+            progress_bar.push(cursor_fg, triangle_poly);
+        }
+
+        use widgetry::DrawWithTooltips;
+        let mut tooltip_text = Text::from(Line("Finished Trips"));
+        tooltip_text.add(Line(format!(
+            "{} ({}% of total)",
+            finished,
+            (ratio * 100.0) as usize
+        )));
+        if let Some(baseline_finished) = self.baseline_finished_trips {
+            // TODO: up/down icons
+            let line = if baseline_finished > finished {
+                let difference = baseline_finished - finished;
+                Line(format!("{} less than baseline", difference))
```

Comment:
`prettyprint_usize` here and below, to help point out really dramatically high differences from the baseline